p2p/protocol/identify/id.go

```diff
@@ -412,6 +409,11 @@ func (ids *IDService) consumeObservedAddress(observed []byte, c inet.Conn) {
 		return
 	}
 
+	if !HasConsistentTransport(c.RemoteMultiaddr(), ids.Host.Addrs()) {
+		log.Debugf("ignoring observed multiaddr that doesn't match the transports of any addresses we're announcing", c.RemoteMultiaddr())
+		return
+	}
```

Comment:
so what happens if we are behind a NAT and advertise a relay addr, and connect to a peer through a different relay (one which we don't advertise)? It seems that we might pollute our advertised set to include this relay too.